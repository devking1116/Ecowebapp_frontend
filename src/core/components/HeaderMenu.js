/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Menu } from 'semantic-ui-react';

type Props = {
  children?: any,
};

/**
 * A menu for the header of the website.
 */
export default function HeaderMenu({ children, ...otherProps }: Props): React.Element<*> {
  const styles = reactCSS({
    'default': {
      menu: {
        height: '100%',
        marginTop: 0,
        marginBottom: 0,
      },
    },
  });

  return (
    <Menu inverted secondary style={styles.menu} {...otherProps}>
    {React.Children.map(children, (child) => child?
      <Menu.Item>
        {child}
      </Menu.Item>:
    null )}
    </Menu>
  );
}
