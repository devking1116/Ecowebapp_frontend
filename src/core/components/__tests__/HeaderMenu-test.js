import React from 'react';
import renderer from 'react-test-renderer';

import HeaderMenu from '../HeaderMenu';

test('it displays its children an inverted secondary semantic-ui Menu.', () => {
  const component = renderer.create(
    <HeaderMenu>
      <div>Test Header Item 0</div>
      <div>Test Header Item 1</div>
      <div>Test Header Item 2</div>
    </HeaderMenu>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
