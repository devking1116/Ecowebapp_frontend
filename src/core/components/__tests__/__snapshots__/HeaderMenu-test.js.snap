exports[`test it displays its children an inverted secondary semantic-ui Menu. 1`] = `
<div
  className="ui inverted secondary menu"
  style={
    Object {
      "height": "100%",
      "marginBottom": 0,
      "marginTop": 0,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div>
      Test Header Item 0
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <div>
      Test Header Item 1
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <div>
      Test Header Item 2
    </div>
  </div>
</div>
`;
